### PR TITLE
[4.0] Check unpublish default workflow

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #21859 .

### Summary of Changes
Fixed the Method publish

### Testing Instructions
Make several workflows, one of them as default.
Then unpublish this workflow

### Expected result
A message appears that this is the default-workflow

### Actual result
The workflow is unpublished

### Documentation Changes Required
no

### CODE REVIEW

File: administrator/language/en-GB/en-GB.com_workflow.ini [hunk 1]
```diff
@@ -91,6 +91,7 @@ COM_WORKFLOW_TRANSITION_DUPLICATE="This transition already exists."
 COM_WORKFLOW_TRANSITION_EDIT="Edit Transition"
 COM_WORKFLOW_TRANSITION_THE_SAME_STAGE="Current stage and target stage are the same."
 COM_WORKFLOW_TRASHED="Trashed"
+COM_WORKFLOW_UNPUBLISH_DEFAULT_ERROR="You try to unpublish the default workflow"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_workflow.ini]: I think a string based on @franz-wohlkoenig suggestion of Can't unpublish default home would be better so try

COM_WORKFLOW_UNPUBLISH_DEFAULT_ERROR="You can't unpublish the default workflow"

**Comment by brianteeman** [comment]
Other than the requested language change this works well

File: administrator/components/com_workflow/Model/WorkflowModel.php [hunk 2]
```diff
@@ -350,21 +350,31 @@ public function publish(&$pks, $value = 1)
 
 		$date = Factory::getDate()->toSql();
 
-		// Default workflow item existence checks.
+		// Default workflow item check.
 		foreach ($pks as $i => $pk)
-		{
-			if ($value != 1 && $table->default)
+		{
```
Comment (richard67) [inline administrator/components/com_workflow/Model/WorkflowModel.php]: Please remove the blank after the `{` in line 355

**Review by brianteeman — APPROVED** [review]
approve language strings

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 354fdb17dfcd1e1cbc41a011ed1d4377886d08fe<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21870">issues.joomla.org/tracker/joomla-cms/21870</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 354fdb17dfcd1e1cbc41a011ed1d4377886d08fe<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21870">issues.joomla.org/tracker/joomla-cms/21870</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**